flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/factory/KubernetesJobManagerFactoryTest.java

```diff
@@ -468,4 +468,37 @@ void testSetJobManagerDeploymentReplicas() throws Exception {
         assertThat(kubernetesJobManagerSpecification.getDeployment().getSpec().getReplicas())
                 .isEqualTo(JOBMANAGER_REPLICAS);
     }
+
+    @Test
+    void testDNSPolicyDefaultValue() throws Exception {
+        kubernetesJobManagerSpecification =
+                KubernetesJobManagerFactory.buildKubernetesJobManagerSpecification(
+                        flinkPod, kubernetesJobManagerParameters);
+
+        final PodSpec resultPodSpecWithDefaultDNSPolicy =
+                this.kubernetesJobManagerSpecification
+                        .getDeployment()
+                        .getSpec()
+                        .getTemplate()
+                        .getSpec();
+
+        assertThat(resultPodSpecWithDefaultDNSPolicy.getDnsPolicy()).isEqualTo("ClusterFirst");
+    }
+
+    @Test
+    void testDNSPolicyOverride() throws Exception {
```

Comment:
This test could be moved to `DecoratorWithPodTemplateTestBase`. And then it will be executed both for `InitJobManagerDecoratorWithPodTemplateTest` and `InitTaskManagerDecoratorWithPodTemplateTest`.

Reply:
Thanks for the suggestion. I'm just not sure what's the best way to overwrite the flink config option?

I was trying something like

```java
    @Test
    void testDNSPolicyOverwritten() {
        this.flinkConfig.set(KubernetesConfigOptions.KUBERNETES_HOSTNETWORK_ENABLED, true);
        assertThat(this.resultPod.getPodWithoutMainContainer().getSpec().getDnsPolicy())
                .isEqualTo(Constants.DNS_POLICY_HOSTNETWORK);
    }
```

but it does not work since flinkConfig is setup during class init.